services/packagist/packagist-dependency-version.service.js

```diff
@@ -0,0 +1,204 @@
+
+const queryParamSchema = Joi.object({
+  server: optionalUrl,
+  version: Joi.string(),
+}).required()
+
+export default class PackagistDependencyVersion extends BasePackagistService {
+  static category = 'platform-support'
+
+  static route = {
+    base: 'packagist/dependency-v',
+    pattern: ':user/:repo/:dependency+',
+    queryParamSchema,
+  }
+
+  static examples = [
+    {
+      title: 'Packagist Dependency Version Support',
+      namedParams: {
+        user: 'symfony',
+        repo: 'symfony',
+        dependency: 'twig/twig',
+      },
+      staticPreview: this.render({
+        dependency: 'twig/twig',
+        dependencyVersion: '2.13|^3.0.4',
+      }),
+    },
+    {
+      title: 'Packagist Dependency Version Support (specify version)',
+      namedParams: {
+        user: 'symfony',
+        repo: 'symfony',
+        dependency: 'twig/twig',
+      },
+      queryParams: {
+        version: 'v2.8.0',
+      },
+      staticPreview: this.render({
+        dependency: 'twig/twig',
+        dependencyVersion: '1.12',
+      }),
+    },
+    {
+      title: 'Packagist Dependency Version Support (custom server)',
+      namedParams: {
+        user: 'symfony',
+        repo: 'symfony',
+        dependency: 'twig/twig',
+      },
+      queryParams: {
+        server: 'https://packagist.org',
+      },
+      staticPreview: this.render({
+        dependency: 'twig/twig',
+        dependencyVersion: '2.13|^3.0.4',
+      }),
+      documentation: customServerDocumentationFragment,
+    },
+    {
+      title: 'Packagist PHP Version Support (custom server)',
+      namedParams: {
+        user: 'symfony',
+        repo: 'symfony',
+        dependency: 'php',
+      },
+      queryParams: {
+        server: 'https://packagist.org',
+        version: 'v2.8.0',
+      },
+      staticPreview: this.render({
+        dependency: 'php',
+        dependencyVersion: '^7.1.3',
+      }),
+      documentation: customServerDocumentationFragment,
+    },
+  ]
+
+  static defaultBadgeData = {
+    label: 'dependency version',
+    color: 'blue',
+  }
+
+  static render({ dependency, dependencyVersion }) {
+    return {
+      label: dependency,
+      message: dependencyVersion,
+    }
+  }
+
+  async getDependencyVersion({
+    json,
+    user,
+    repo,
+    dependency,
+    version = '',
+    server,
+  }) {
+    let packageVersion
+    const versions = BasePackagistService.expandPackageVersions(
+      json,
+      this.getPackageName(user, repo)
+    )
+
+    if (version === '') {
+      packageVersion = this.findLatestRelease(versions)
+    } else {
+      try {
+        packageVersion = await this.findSpecifiedVersion(
+          versions,
+          user,
+          repo,
+          version,
+          server
+        )
+      } catch (e) {
+        packageVersion = null
+      }
+    }
+
+    if (!dependency) {
+      throw new NotFound({
+        prettyMessage: 'dependency vendor or repo not specified',
+      })
+    }
```

Comment:
Do we need this check? I think `:dependency+` should enforce this constraint at the routing layer

Reply:
You’re right, thanks, I’ll remove this check.